core/clustersmngr/client.go

```diff
@@ -103,6 +111,13 @@ func (c *clustersClient) List(ctx context.Context, cluster string, list client.O
 		return err
 	}
 
+	// Due to how DelegatingClients work, calls that fails never returns,
+	// because it waits the cache to sync before returning it https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/internal/informers_map.go#L206
+	// so we are forced to use a timeout so it doesn't keep the informer up.
+	// TODO: What should be the timeout here?
+	ctx, cancel := context.WithTimeout(ctx, clientTimeout)
```

Comment:
can you expand a bit more on the issue to understand it in order to input the timeout value?

Reply:
I see now, thanks for the explanation  ... it now makes sense to me

follow up question - is the cache always filled when empty whenever is a get or a list operation?

Reply:
so we have three scenarios 

1. access allowed and cache hot  -> quick answer
2. access allowed and cache cold -> up to the size of the collection 
3. not allowed -> not answer -> timeout

what would be the result for the user when in scenario 2) and timeout = 2sec, to fill up the cache takes more than 2 sec? there would be a timeout and then is there any retry or the client see an error?

I could think that a strategy of failing fast + retry could be a good balance

Reply:
to me sounds sensible, we could enhance this once tested with Protocol Labs if we see that is not a great experience